### PR TITLE
Refactoring Tabs

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

**Review by OsamaAbdellateef — COMMENTED** [review]
Reviewed

File: packages/components/src/Tabs/Tab.tsx [hunk 1]
```diff
@@ -19,25 +19,22 @@ export const Tab = ({ item, state }: TabProps) => {
   const { tabProps, isSelected } = useTab({ key }, state, ref);
   const disabled = tabProps['aria-disabled'];
   const { hoverProps, isHovered } = useHover({
-    isDisabled: disabled as boolean,
+    isDisabled: (disabled as boolean) || isSelected,
```
Comment (sarahgm) [inline packages/components/src/Tabs/Tab.tsx:22]: @OsamaAbdellateef why also `isSelected` ?

File: packages/components/src/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -1,60 +1,57 @@
...
-import { TabContext } from './Context';
 import { Item } from '@react-stately/collections';
+import { TabContext } from './Context';
 
-export interface TabsThemeExtension
-  extends ThemeExtensionsWithParts<'Tabs', ['tabs', 'tab', 'tabPanel']> {}
 interface TabsProps
   extends Omit<AriaTabListProps<object>, 'orientation' | 'isDisabled'> {
-  gap?: number;
+  gap?: string;
```
Comment (sarahgm) [inline packages/components/src/Tabs/Tabs.tsx:15]: you can use for this also `gapSpace` which is in `style-props.tsx`

File: packages/components/src/Tabs/Tab.tsx (hunk 1 above)
Comment (OsamaAbdellateef) [reply packages/components/src/Tabs/Tab.tsx:22]: i wanna disable the hover if the tab is `selected` 
i guess it shouldn't have `hover` effect when it is already `selected`

File: packages/components/src/Tabs/Tabs.tsx (hunk 2 above)
Comment (OsamaAbdellateef) [reply packages/components/src/Tabs/Tabs.tsx]: Yeah it is better, now it is added

**Review by sarahgm — APPROVED** [review]

**Pull request merged**